src/wmtk/TriMesh.h

```diff
@@ -311,7 +310,22 @@ class TriMesh
 protected:
     virtual bool split_before(const Tuple& t) { return true; }
     virtual bool split_after(const Tuple& t) { return true; }
-    virtual bool collapse_before(const Tuple& t) { return true; }
+    // check link, check if it's the last edge
+    virtual bool collapse_before(const Tuple& t)
+    {
+        auto v1_conn_tris = m_vertex_connectivity[t.get_vid()].m_conn_tris;
+        auto v2_conn_tris = m_vertex_connectivity[switch_vertex(t).get_vid()].m_conn_tris;
+        size_t fid1 = t.get_fid();
+        Tuple tmp_tuple = switch_face(t).value_or(t);
+        size_t fid2 = tmp_tuple.get_fid();
+        vector_erase(v1_conn_tris, fid1);
+        vector_erase(v1_conn_tris, fid2);
+        vector_erase(v2_conn_tris, fid1);
+        vector_erase(v2_conn_tris, fid2);
```

Comment:
what is all of this?

Reply:
I set the constraint that if all the triangles left depend on the edge to be collapsed then this edge cannot be collapsed. Basically, after all the collapsing there has to be at least one triangle left

Reply:
I see now. This seems to be equivalent to, and maybe not necesary.
```
if (v1_conn_tris.size() == 2 && v2_conn_tris.size() == 2 )
```

Reply:
it's equivalent to if (v1_conn_tris.size() == 2 || v2_conn_tris.size() == 2 )

Reply:
Then why not just write that?